packages/python-google-compute-engine/google_compute_engine/networking/network_setup/network_setup.py

```diff
@@ -50,16 +51,38 @@ def EnableIpv6(self, interfaces):
     Args:
       interfaces: list of string, the output device names for enabling IPv6.
     """
-    if not interfaces or set(interfaces) == self.ipv6_interfaces:
+    if not interfaces or self.ipv6_interfaces == set(interfaces):
       return
 
     self.logger.info('Enabling IPv6 on Ethernet interface: %s.', interfaces)
-    self.ipv6_interfaces = set(interfaces)
+    self.ipv6_interfaces = self.ipv6_interfaces.union(set(interfaces))
+    self.ipv6_interfaces_initialized = True
```

Comment:
Can we shorted this variable to `ipv6_initialized`. Also, when is this variable ever false? It appears that you always check the set intersection as well as this variable. Set intersection should be enough.

Reply:
Renamed var, it is set to False in _init also intersection is not sufficient condition. Since we want the code to run at least once.